### PR TITLE
minor: Kill Mutation for AbstractHeaderCheck

### PR DESCRIPTION
**Pull request by Kevin222004**
minor: Kill Mutation for  AbstractHeaderCheck

for  https://github.com/checkstyle/checkstyle/pull/13269

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java [hunk 1]
```diff
@@ -380,4 +383,30 @@ public void testNoHeaderMissingErrorInCaseHeaderSizeEqualToFileSize() throws Exc
...
 
+    @Test
+    public void testCharsetProperty() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(RegexpHeaderCheck.class);
+        checkConfig.addProperty("headerFile", getPath("InputRegexpHeader6.header"));
+        Checker checker = createChecker(checkConfig);
+        checker.setCharset("US-ASCII");
+        final String[] expected = {
+            "1: " + getCheckMessage(MSG_HEADER_MISMATCH, "// some.class.тест.passed"),
+        };
```
Comment (Kevin222004) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java:394]: @romani in the same file I have created 2 test cases with name `testCharsetProperty` and `testCharsetProperty2`
To show that the setter in AbstractHeaderCheck is useless.

In first test case `testCharsetProperty` I have set the property `charset` in the parent module checker. 
hence you can observe the violation 
`"1: " + getCheckMessage(MSG_HEADER_MISMATCH, "// some.class.тест.passed"),` 


and in second Test case `testCharsetProperty2` where property is setted in check it self 
and the message we got 
`"1: " + getCheckMessage(MSG_HEADER_MISMATCH, "// some.class.��������.passed"),` 

so

Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java:394]: I am not sure about this that why this is happening, I have currently removed the setter from the main pr.

Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java:394]: @romani please give a look

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java:394]: please create same testing by CLI, to make sure we do all as users and system works as expected.
in junit we always need to be accurate as we might miss some actions that actual system is doing.
We did propogation of such properties to all children modules.

Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java:394]:
```
kevin@inspiron-15-5510:~/Desktop/check_style/header$ cat config.header 
// some.class.тест.passed
kevin@inspiron-15-5510:~/Desktop/check_style/header$ cat header.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
                "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
        <property name="charset" value="US-ASCII" />
        <module name="RegexpHeader">
                <property name="headerFile" value="config.header"/>
        </module>
</module>
kevin@inspiron-15-5510:~/Desktop/check_style/header$ cat Test2.java 
// some.class

public class Test2 {
}
kevin@inspiron-15-5510:~/Desktop/check_style/header$ 
kevin@inspiron-15-5510:~/Desktop/check_style/header$ java -jar /home/kevin/Downloads/checkstyle-10.12.1-all.jar -c header.xml Test2.java
Starting audit...
[ERROR] /home/kevin/Desktop/check_style/header/Test2.java:1: Line does not match expected header line of '// some.class.��������.passed'. [RegexpHeader]
Audit done.
Checkstyle ends with 1 errors.
```

-------------------

```
kevin@inspiron-15-5510:~/Desktop/check_style/header$ cat config.header 
// some.class.тест.passed
kevin@inspiron-15-5510:~/Desktop/check_style/header$ cat Test2.java 
// some.class

public class Test2 {
}
kevin@inspiron-15-5510:~/Desktop/check_style/header$ cat header2.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
                "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
        <!-- <property name="charset" value="US-ASCII" /> -->
        <module name="RegexpHeader">
                <property name="charset" value="US-ASCII" />
                <property name="headerFile" value="config.header"/>
        </module>
</module>

kevin@inspiron-15-5510:~/Desktop/check_style/header$ java -jar /home/kevin/Downloads/checkstyle-10.12.1-all.jar -c header2.xml Test2.java
Starting audit...
[ERROR] /home/kevin/Desktop/check_style/header/Test2.java:1: Line does not match expected header line of '// some.class.��������.passed'. [RegexpHeader]
Audit done.
Checkstyle ends with 1 errors.
kevin@inspiron-15-5510:~/Desktop/check_style/header$ 
```

Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java:394]: @romani  looks like with cli non of them is working

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java:394]: If we drop non ASCII symbols, expected match is `// some.class..passed`
Please create Input based test method to see what is actual string for match.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java:392]: @Kevin222004 , ping

Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java]: @romani I have some misconceptions on this pr.

**Comment by Kevin222004** [comment]
I have updated the pr and updated the test as well and they are working as expected. `тест` this text will be accepted by charset UTF_8 and not expected by `US-ASCII`

Just for mutation purpose the mutation on setCharset is killed https://github.com/checkstyle/checkstyle/blob/08276244664c6e44f267fb5cc861c208d14591cb/config/pitest-suppressions/pitest-header-suppressions.xml#L39-L46

But still I believe that, the actuall testing of this property is been done when the issue  https://github.com/checkstyle/checkstyle/issues/13392
will be resolved and we set some different charset in checker and some other charset in our check.

**Comment by Kevin222004** [comment]
Rest all the mutations https://github.com/checkstyle/checkstyle/blob/08276244664c6e44f267fb5cc861c208d14591cb/config/pitest-suppressions/pitest-header-suppressions.xml#L2-L37

are on the line https://github.com/checkstyle/checkstyle/blob/08276244664c6e44f267fb5cc861c208d14591cb/src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java#L63-L64

In my opinion this is by default `UTF_8` so we can simple Change this as 
`private Charset charset;`

**Comment by Kevin222004** [comment]
@romani ping 
https://github.com/checkstyle/checkstyle/pull/13268#issuecomment-1645989639 
https://github.com/checkstyle/checkstyle/pull/13268#issuecomment-1645993047

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java (hunk 1 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java]: @Kevin222004 , please update this PR to have mutation removal, I lost a point of this PR.

**Comment by romani** [comment]
I confirm that 
initialization https://github.com/checkstyle/checkstyle/blob/08276244664c6e44f267fb5cc861c208d14591cb/src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java#L63-L64 should be removed as we set all charsets from Checker:
https://github.com/checkstyle/checkstyle/blob/08276244664c6e44f267fb5cc861c208d14591cb/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L123 
and populated further to all modules by context:
https://github.com/checkstyle/checkstyle/blob/08276244664c6e44f267fb5cc861c208d14591cb/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L448-L455

so after module instantiation, this Check use charset from its itialization, but it instantly changed by setter to what ever is Checker/context.

**Pull request closed without merging**